bundles/org.openhab.binding.herzborg/src/main/java/org/openhab/binding/herzborg/internal/SerialBus.java

```diff
@@ -0,0 +1,134 @@
+ */
+public class SerialBus extends Bus implements SerialPortEventListener {
+    private final Logger logger = LoggerFactory.getLogger(SerialBus.class);
+    private SerialPortManager serialPortManager;
+    private @Nullable SerialPort serialPort;
+
+    public SerialBus(SerialPortManager manager) {
+        serialPortManager = manager;
+    }
+
+    public Result initialize(String port) {
+        SerialPortIdentifier portIdentifier = serialPortManager.getIdentifier(port);
+        if (portIdentifier == null) {
+            return new Result(ThingStatusDetail.CONFIGURATION_ERROR, "No such port: " + port);
+        }
+
+        SerialPort commPort;
+        try {
+            commPort = portIdentifier.open(this.getClass().getName(), 2000);
+        } catch (PortInUseException e1) {
+            return new Result(ThingStatusDetail.CONFIGURATION_ERROR, "Port " + port + " is in use");
+        }
+
+        try {
+            // Herzborg serial bus operates with fixed parameters
+            commPort.setSerialPortParams(9600, SerialPort.DATABITS_8, SerialPort.STOPBITS_1, SerialPort.PARITY_NONE);
+            commPort.setFlowControlMode(SerialPort.FLOWCONTROL_NONE);
+        } catch (UnsupportedCommOperationException e) {
+            return new Result(ThingStatusDetail.CONFIGURATION_ERROR, "Invalid port configuration");
+        }
+
+        try {
+            commPort.enableReceiveThreshold(8);
+            commPort.enableReceiveTimeout(1000);
+        } catch (UnsupportedCommOperationException e) {
+            // OpenHAB's serial-over-IP doesn't support these, so let's ignore the exception
+        }
+
+        InputStream dataIn = null;
+        OutputStream dataOut = null;
+        String error = null;
+
+        try {
+            dataIn = commPort.getInputStream();
+            dataOut = commPort.getOutputStream();
+
+            if (dataIn == null) {
+                error = "No input stream available on the serial port";
+            } else if (dataOut == null) {
+                error = "No output stream available on the serial port";
+            } else {
+                dataOut.flush();
+                if (dataIn.markSupported()) {
+                    dataIn.reset();
+                }
+
+                // RXTX serial port library causes high CPU load
+                // Start event listener, which will just sleep and slow down event
+                // loop
+                // This is copied from SonyProjector binding
+                commPort.addEventListener(this);
+                commPort.notifyOnDataAvailable(true);
+            }
+        } catch (IOException | TooManyListenersException e) {
+            error = e.getMessage();
+        }
+
+        if (error != null) {
+            return new Result(ThingStatusDetail.HANDLER_INITIALIZING_ERROR, error);
+        }
+
+        this.serialPort = commPort;
+        this.dataIn = dataIn;
+        this.dataOut = dataOut;
+
+        logger.trace("Successfully initialized");
```

Comment:
Can this be removed if the thing comes online right away it will go into the event.log automatically?